src/Microsoft.AspNetCore.Mvc.ApiExplorer/ApiResponseTypeProvider.cs

```diff
@@ -172,6 +176,15 @@ private static List<IApiResponseMetadataProvider> GetResponseMetadataAttributes(
                             });
                         }
                     }
+
+                    if (!isSupportedContentType && contentType != null)
+                    {
+                        // No output formatter was found that supports this content type. Add the user specified content type as-is to the result.
+                        apiResponse.ApiResponseFormats.Add(new ApiResponseFormat
+                        {
+                            MediaType = contentType,
+                        });
+                    }
```

Comment:
This is kinda a roundabout way of looking at it I guess. Is the idea to preserve the mapping of content type->formatter?